### DISCUSSION TITLE
Observing Rx Packets Drop on tunnel interface

### DISCUSSION THREAD

**Discussion by jayaprakashsajja** [strongswan/strongswan#792] [Q&A]
Rx packets are getting dropped on RX , but not huge.
The percent of receiving packets loss is around 0.33% but iam loosing packets which are Important.
        RX packets 6980952160  bytes 7804078538611 (7.0 TiB)
        RX errors 0  dropped 23716972  overruns 0  frame 0

**Comment by tobiasbrunner** [comment]
You realize IPsec, just like IP, is an unreliable protocol with best-effort delivery, right? If you need reliability, use TCP or something similar on top.

Having said that, dropped packets without errors seems weird (at least for VTI/XFRM interfaces, don't know what kind of interface you are referring to exactly, though). Might indicate some buffer issue.

**Reply by jayaprakashsajja** [reply]
Can you pls throw us a light on how to use TCP on top of IP, and will it cause any delay in processing ?
And also to increase the buffer size as we are using optimum buffer size.
cat /proc/sys/net/ipv4/tcp_rmem
4096    16777216        16777216

**Reply by tobiasbrunner** [reply]
> Can you pls throw us a light on how to use TCP on top of IP, and will it cause any delay in processing ?

I'm not suggesting to stuff generic IP packets in a TCP stream, but use TCP or some other reliable protocol to transport whatever it is you want to transport. What transport protocol did the lost packets use anyway? UDP?

Increasing the TCP buffer would only be useful if you'd actually use TCP. There are others for e.g. UDP or receive buffers on individual sockets and there might even be more generic ones (no idea, really).
